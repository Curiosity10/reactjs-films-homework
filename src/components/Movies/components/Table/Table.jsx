import React from 'react';
import TableCard from './components/TableCard';

const Table = ({ movies }) => (movies.map(movie => (
  <TableCard
    key={movie.id}
    movie={movie}
  />
))
);

export default Table;
