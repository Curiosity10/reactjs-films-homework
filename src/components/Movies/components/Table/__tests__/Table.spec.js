import React from 'react';
import { create } from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import Table from '..';

describe('Table component renders correctly', () => {
  const movies = [
    {
      title: 'test',
      id: 1,
      genre_ids: [
        {
          id: 12,
          name: 'Action',
        },
        {
          id: 16,
          name: 'Adventure',
        },
      ],
      overview: 'desc',
    },
    {
      title: 'test2',
      id: 2,
      genre_ids: [
        {
          id: 21,
          name: 'Action',
        },
        {
          id: 23,
          name: 'Adventure',
        },
      ],
      overview: 'desc2',
    },
  ];
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const initialState = {
    app: {
      movies: [],
      genres: [],
      isLoading: true,
      currentPage: 1,
      error: false,
      filter: 'Trending',
      genreId: null,
      searchQuery: '',
      videoKey: '',
      pagesCount: 1,
    },
    ui: {
      layout: 'table',
    },
  };
  const store = mockStore(initialState);

  it('Table renders correctly', () => {
    const tree = create(
      <Provider store={store}>
        <Table movies={movies} />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
