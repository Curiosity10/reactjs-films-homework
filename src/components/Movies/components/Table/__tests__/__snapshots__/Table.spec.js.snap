// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component renders correctly Table renders correctly 1`] = `
Array [
  <article
    className="card"
  >
    <img
      alt="Poster"
    />
    <div
      className="info"
    >
      <h2
        className="title"
      >
        <a
          href="/movie/:1"
          onClick={[Function]}
        >
          test
        </a>
      </h2>
      <p
        className="overview"
      >
        desc
      </p>
      <div
        className="rating"
      />
      <button
        aria-label="Watch Now"
        className="btn primary"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
    />
    <div
      className="info"
    >
      <h2
        className="title"
      >
        <a
          href="/movie/:2"
          onClick={[Function]}
        >
          test2
        </a>
      </h2>
      <p
        className="overview"
      >
        desc2
      </p>
      <div
        className="rating"
      />
      <button
        aria-label="Watch Now"
        className="btn primary"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
    </div>
  </article>,
]
`;
